levelZeroLib/src/levelZeroPowerMonitor.cpp

```diff
@@ -0,0 +1,274 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "levelZeroPowerMonitor.h"
+
+#include <iostream>
+#include <vector>
+
+
+#include "ze_api.h"
+#include "ze_log.h"
+#include "zes_api.h"
+
+#define VALIDATECALL(myZeCall) \
+    if (myZeCall != ZE_RESULT_SUCCESS){ \
+        std::cout << "Error at "       \
+            << #myZeCall << ": "       \
+            << __FUNCTION__ << ": "    \
+            << __LINE__ << std::endl;  \
+        std::cout << "Exit with Error Code: " \
+            << "0x" << std::hex \
+            << myZeCall \
+            << std::dec << std::endl; \
+    }
```

Comment:
We already have the `ze_log.h` file that logs and prints similar info. Is this call different?

If not, I think we can reuse the function that we have, and also trigger the reporting at the JNI level only if the `LOG_JNI` flag is defined (also in the `ze_log.h` file). Otherwise, we will end up reporting the messages and impact the time spent at the native level.